closure_tree/migrations.py

```diff
@@ -1,6 +1,28 @@
 from django.db.migrations.operations.special import RunSQL
 
 
+VIEW_TEMPLATE = '''
+CREATE OR REPLACE RECURSIVE VIEW {view}(path, ancestor_id, descendant_id, depth) AS
+
+SELECT ARRAY[{node_id}], {node_id}, {node_id}, 0
+FROM {table}
+
+UNION ALL
+
+SELECT {table}.{parent_id} || {view}.path, {table}.{parent_id}, {view}.descendant_id, {view}.depth + 1
+FROM {table}
+INNER JOIN {view} ON ({view}.ancestor_id = {table}.{node_id})
+WHERE {table}.{parent_id} IS NOT NULL;
+'''
+
+
+def get_self_reference(model):
+    parent_id = [field.column for field in model._meta.fields if field.rel and field.rel.to is model]
+    if len(parent_id) != 1:
+        raise Exception('Unable to detect self-referential field.')
```

Comment:
Perhaps separate messages for 0 vs too many?

Reply:
Also I guess some mechanism to supply which field it is when there are multiple.